rules/Php54/Rector/Array_/ArrayToShortArrayRector.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Php54\Rector\Array_;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Array_;
+use Rector\Core\Provider\CurrentFileProvider;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\ValueObject\Application\File;
+use Rector\Core\ValueObject\PhpVersionFeature;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\Php54\Rector\Array_\ArrayToShortArrayRector\ArrayToShortArrayRectorTest
+ */
+final class ArrayToShortArrayRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function __construct(private readonly CurrentFileProvider $currentFileProvider)
+    {
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::SHORT_ARRAY;
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Array to short array',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        return array();
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        return [];
+    }
+}
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Array_::class];
+    }
+
+    /**
+     * @param Array_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $file = $this->currentFileProvider->getFile();
+        if (! $file instanceof File) {
+            return null;
+        }
+
+        $oldTokens = $file->getOldTokens();
+        $startTokenPos = $node->getStartTokenPos();
+        if (! isset($oldTokens[$startTokenPos][1])) {
+            return null;
+        }
+
+        $arrayStart = $oldTokens[$startTokenPos][1];
+        if ($arrayStart === '[') {
+            return null;
+        }
+
+        return new Array_($node->items);
```

Comment:
Reprint the Node should make it printed as short array:

```php
$node->setAttribute( \Rector\NodeTypeResolver\Node\AttributeKey::ORIGINAL_NODE, null);
return $node;
```

Reply:
This doesn't seem to work; the `array()` syntax is still used when the node is printed. I'm not exactly sure why.

Reply:
It seems there is KIND_LONG/SHORT that can be used, ref 

https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/Node/Expr/Array_.php

You can set AttributeKey::KIND, Array_::KIND_SHORT for it, checking long can use Array_::KIND_LONG

Reply:
Ref https://github.com/rectorphp/rector-src/blob/f608461e44c6a4008a2bff6c603ba9d52d3ffacf/src/PhpParser/Printer/BetterStandardPrinter.php#L306

Reply:
Looks like it works if I also add:

```
$node->setAttribute(\Rector\NodeTypeResolver\Node\AttributeKey::KIND, Array_::KIND_SHORT);
```

With this, I think I can also check the attribute instead of looking at the tokens, so I'll change that. 👍